In [1]:
#data handling/modeling
import pandas as pd
import numpy as np
import pylab
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn import metrics
import scipy.stats as stats

# visualization
%matplotlib inline
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
df=pd.read_csv('C:/Users/Operator/Desktop/dataAnalysis/archive/Real estate.csv')
df

No  X1 transaction date  X2 house age  \
0      1             2012.917          32.0   
1      2             2012.917          19.5   
2      3             2013.583          13.3   
3      4             2013.500          13.3   
4      5             2012.833           5.0   
..   ...                  ...           ...   
409  410             2013.000          13.7   
410  411             2012.667           5.6   
411  412             2013.250          18.8   
412  413             2013.000           8.1   
413  414             2013.500           6.5   

     X3 distance to the nearest MRT station  X4 number of convenience stores  \
0                                  84.87882                               10   
1                                 306.59470                                9   
2                                 561.98450                                5   
3                                 561.98450                                5   
4                                 390.56840                                5   
..                                      ...                              ...   
409                              4082.01500                                0   
410                                90.45606                                9   
411                               390.96960                                7   
412                               104.81010                                5   
413                                90.45606                                9   

     X5 latitude  X6 longitude  Y house price of unit area  
0       24.98298     121.54024                        37.9  
1       24.98034     121.53951                        42.2  
2       24.98746     121.54391                        47.3  
3       24.98746     121.54391                        54.8  
4       24.97937     121.54245                        43.1  
..           ...           ...                         ...  
409     24.94155     121.50381                        15.4  
410     24.97433     121.54310                        50.0  
411     24.97923     121.53986                        40.6  
412     24.96674     121.54067                        52.5  
413     24.97433     121.54310                        63.9  

[414 rows x 8 columns]

In [3]:
df.isnull().sum()

No                                        0
X1 transaction date                       0
X2 house age                              0
X3 distance to the nearest MRT station    0
X4 number of convenience stores           0
X5 latitude                               0
X6 longitude                              0
Y house price of unit area                0
dtype: int64

In [4]:
madrid= pd.read_csv('C:/Users/Operator/Desktop/dataAnalysis/archive/archive/houses_Madrid.csv')
madrid


Unnamed: 0     id                                              title  \
0               0  21742              Piso en venta en calle de Godella, 64   
1               1  21741   Piso en venta en calle de la del Manojo de Rosas   
2               2  21740               Piso en venta en calle del Talco, 68   
3               3  21739               Piso en venta en calle Pedro Jiménez   
4               4  21738  Piso en venta en carretera de Villaverde a Val...   
...           ...    ...                                                ...   
21737       21737      5       Ático en venta en Casco Histórico de Barajas   
21738       21738      4  Piso en venta en Urb. Campo de las Naciones - ...   
21739       21739      3  Piso en venta en Campo de las Naciones-Corralejos   
21740       21740      2      Chalet adosado en venta en calle Burriana, 12   
21741       21741      1             Piso en venta en calle San Epifanio, 9   

                                       subtitle  sq_mt_built  sq_mt_useful  \
0                         San Cristóbal, Madrid         64.0          60.0   
1                           Los Ángeles, Madrid         70.0           NaN   
2                            San Andrés, Madrid         94.0          54.0   
3                            San Andrés, Madrid         64.0           NaN   
4                           Los Rosales, Madrid        108.0          90.0   
...                                         ...          ...           ...   
21737                           Barajas, Madrid         78.0           NaN   
21738                           Barajas, Madrid         96.0          83.0   
21739                           Barajas, Madrid        175.0           NaN   
21740  Campo de las Naciones-Corralejos, Madrid        289.0           NaN   
21741                          Imperial, Madrid         72.0           NaN   

       n_rooms  n_bathrooms  n_floors  sq_mt_allotment  ...  \
0            2          1.0       NaN              NaN  ...   
1            3          1.0       NaN              NaN  ...   
2            2          2.0       NaN              NaN  ...   
3            2          1.0       NaN              NaN  ...   
4            2          2.0       NaN              NaN  ...   
...        ...          ...       ...              ...  ...   
21737        2          2.0       NaN              NaN  ...   
21738        2          2.0       NaN              NaN  ...   
21739        4          2.0       NaN              NaN  ...   
21740        4          3.0       3.0            304.0  ...   
21741        2          2.0       NaN              NaN  ...   

       energy_certificate  has_parking has_private_parking  \
0                       D        False                 NaN   
1              en trámite        False                 NaN   
2             no indicado        False                 NaN   
3              en trámite        False                 NaN   
4              en trámite         True                 NaN   
...                   ...          ...                 ...   
21737          en trámite         True                 NaN   
21738          en trámite         True                 NaN   
21739                   E         True                 NaN   
21740          en trámite         True                 NaN   
21741                   A         True                 NaN   

       has_public_parking is_parking_included_in_price parking_price  \
0                     NaN                          NaN           NaN   
1                     NaN                          NaN           NaN   
2                     NaN                          NaN           NaN   
3                     NaN                          NaN           NaN   
4                     NaN                         True           0.0   
...                   ...                          ...           ...   
21737                 NaN                         True           0.0   
21738                 NaN                  

In [6]:
madrid.columns

Index(['Unnamed: 0', 'id', 'title', 'subtitle', 'sq_mt_built', 'sq_mt_useful',
       'n_rooms', 'n_bathrooms', 'n_floors', 'sq_mt_allotment', 'latitude',
       'longitude', 'raw_address', 'is_exact_address_hidden', 'street_name',
       'street_number', 'portal', 'floor', 'is_floor_under', 'door',
       'neighborhood_id', 'operation', 'rent_price', 'rent_price_by_area',
       'is_rent_price_known', 'buy_price', 'buy_price_by_area',
       'is_buy_price_known', 'house_type_id', 'is_renewal_needed',
       'is_new_development', 'built_year', 'has_central_heating',
       'has_individual_heating', 'are_pets_allowed', 'has_ac',
       'has_fitted_wardrobes', 'has_lift', 'is_exterior', 'has_garden',
       'has_pool', 'has_terrace', 'has_balcony', 'has_storage_room',
       'is_furnished', 'is_kitchen_equipped', 'is_accessible',
       'has_green_zones', 'energy_certificate', 'has_parking',
       'has_private_parking', 'has_public_parking',
       'is_parking_included_in_price', 'parki

In [8]:
madrid.isnull().sum()

Unnamed: 0                          0
id                                  0
title                               0
subtitle                            0
sq_mt_built                       126
sq_mt_useful                    13514
n_rooms                             0
n_bathrooms                        16
n_floors                        20305
sq_mt_allotment                 20310
latitude                        21742
longitude                       21742
raw_address                      5465
is_exact_address_hidden             0
street_name                      5905
street_number                   15442
portal                          21742
floor                            2607
is_floor_under                   1170
door                            21742
neighborhood_id                     0
operation                           0
rent_price                          0
rent_price_by_area              21742
is_rent_price_known                 0
buy_price                           0
buy_price_by

columns to be removed

High Missingness: Columns with a very high proportion of missing values, where imputation might not be meaningful or feasible.

n_floors: 20,305 missing values
sq_mt_allotment: 20,310 missing values
latitude & longitude: 21,742 missing values
portal, door: 21,742 missing values
rent_price_by_area: 21,742 missing values
are_pets_allowed: 21,742 missing values
is_furnished, is_kitchen_equipped: 21,742 missing values
has_private_parking, has_public_parking: 21,742 missing values


Potentially Redundant or Less Significant Columns: Columns that might not carry distinct information or may not be critical for the analysis.

raw_address: If you're already using street_name and neighborhood_id, this might be redundant.
street_number: 15,442 missing values and might not be crucial for many types of analyses

Low Missingness: Columns with a low proportion of missing values, where imputation might be meaningful or feasible.

Columns with Moderate Missingness: You could either drop these columns or consider imputation based on the nature of the column and the analysis's objectives.

built_year: 11,742 missing values
has_ac: 10,531 missing values
has_fitted_wardrobes: 8,343 missing values
has_central_heating, has_individual_heating: 8,134 missing values
Columns related to property orientation: Over 10,384 missing values each
Columns with Some Missingness: You might want to retain these columns and consider imputation methods or domain knowledge to fill the missing values.

n_bathrooms: 16 missing values
house_type_id: 391 missing values
is_new_development: 992 missing values
has_lift: 2,386 missing values
is_exterior: 3,043 missing values